llvm/lib/Target/AMDGPU/AMDGPUInstCombineIntrinsic.cpp

```diff
@@ -1378,6 +1380,8 @@ std::optional<Value *> GCNTTIImpl::simplifyDemandedVectorEltsIntrinsic(
   case Intrinsic::amdgcn_raw_ptr_buffer_load:
   case Intrinsic::amdgcn_raw_buffer_load_format:
   case Intrinsic::amdgcn_raw_ptr_buffer_load_format:
+  case Intrinsic::amdgcn_raw_atomic_buffer_load:
+  case Intrinsic::amdgcn_raw_atomic_ptr_buffer_load:
```

Comment:
Needs instcombine tests for this. Also, would be best to do this in a separate change from the introduction of the intrinsic

Reply:
In theory, the test with the `<4 x i32>` load should show the effects, since only one element is used, no? But adding the intrinsics to this unfortunately did not have any effect. When I tried debugging with lldb, it seemed like this method was never called. If there is anything missing, I'd love to know!